### PR TITLE
CVE summaries system test implementation

### PR DESCRIPTION
**Pull request by rcohencyberarmor**
## PR Type:
Tests

___
## PR Description:
This PR introduces system tests for the CVE summaries feature. It includes changes in the test files to validate the expected CVE summaries and namespace CVE summaries. The PR also includes the addition of expected result JSON files for the CVE summaries.

___
## PR Main Files Walkthrough:
`configurations/system/tests_cases/relevant_vuln_scanning_tests.py`: Added expected_CVEs_summaries and expected_namespace_CVEs_summaries to the test cases.
`tests_scripts/helm/base_vulnerability_scanning.py`: Added methods to store and validate CVE summaries and namespace CVE summaries for the first time results.
`tests_scripts/kubernetes/base_k8s.py`: Added methods to get CVE summaries and namespace CVE summaries from the storage.
`tests_scripts/helm/relevant_cve.py`: Added steps in the start method to validate the expected CVE summaries and namespace CVE summaries.
`systest_utils/statics.py`: Added constants for CVE summaries and namespace CVE summaries.
`configurations/relevant_cves/expected-result/wikijs/CVESSummaries/nginx.json`: Added an expected result JSON file for the CVE summary of nginx.
`configurations/relevant_cves/expected-result/wikijs/namespaceCVESSummaries/wikijs.json`: Added an expected result JSON file for the namespace CVE summary of wikijs.

### CODE REVIEW

File: tests_scripts/kubernetes/base_k8s.py [hunk 1]
```diff
@@ -1068,3 +1068,52 @@ def get_filtered_CVEs_from_storage(self, filteredCVEsKEys):
             )
               filteredCVEs.append((key, cve_data))
         return filteredCVEs
+
+    def get_CVEs_summaries_from_storage(self, CVEsSummariesKeys, namespace):
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/kubernetes/base_k8s.py:1072]:
```suggestion
    def get_CVEs_summaries_from_storage(self, CVEs_summaries_keys, namespace):
```

File: tests_scripts/kubernetes/base_k8s.py [hunk 2]
```diff
@@ -1068,3 +1068,52 @@ def get_filtered_CVEs_from_storage(self, filteredCVEsKEys):
...
+        CVEsSummaries = []
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/kubernetes/base_k8s.py:1073]:
```suggestion
        CVEs_summaries = []
```

File: tests_scripts/kubernetes/base_k8s.py [hunk 3]
```diff
@@ -1068,3 +1068,52 @@ def get_filtered_CVEs_from_storage(self, filteredCVEsKEys):
...
+        if any(isinstance(i, list) for i in CVEsSummariesKeys):
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/kubernetes/base_k8s.py:1074]: why do we need this if? From your implementation of `get_CVEs_summaries_keys` it looks like you know the value being returned?

File: tests_scripts/kubernetes/base_k8s.py [hunk 4]
```diff
@@ -1068,3 +1068,52 @@ def get_filtered_CVEs_from_storage(self, filteredCVEsKEys):
...
+            for keys in CVEsSummariesKeys:
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/kubernetes/base_k8s.py:1075]: let's follow python convention and use pascal case

File: tests_scripts/kubernetes/base_k8s.py [hunk 5]
```diff
@@ -1068,3 +1068,52 @@ def get_filtered_CVEs_from_storage(self, filteredCVEsKEys):
...
+                version=statics.STORAGE_AGGREGATED_API_VERSION,
+                name=key,
+                namespace=namespace,
+                plural=statics.STORAGE_CVES_SUMMARIES_PLURAL,
+            )
+                CVEsSummaries.append((key, CVE_data))
+        return CVEsSummaries
+
+
+    def get_CVEs_summaries_by_namespace_from_storage(self, CVEsSummariesKeys):
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/kubernetes/base_k8s.py:1098]: this name is a bit confusing. If the function "knows" that the keys are actually namespace names, so lets receive a parameter named `namespaces`. If you want to abstract away this detail from the function and call the parameter `CVEsSummariesKeys`, so let's not have  "by_namespace" on the name.
IMO it should be called `get_CVEs_summaries_by_scope_from_storage`, and the scope is decided as part of the key generation

File: tests_scripts/kubernetes/base_k8s.py [hunk 6]
```diff
@@ -1068,3 +1068,52 @@ def get_filtered_CVEs_from_storage(self, filteredCVEsKEys):
...
+        CVEsSummaries = []
+        if any(isinstance(i, list) for i in CVEsSummariesKeys):
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/kubernetes/base_k8s.py:1100]: again, why do we need this if?

File: tests_scripts/helm/base_vulnerability_scanning.py [hunk 7]
```diff
@@ -1035,6 +1065,49 @@ def validate_expected_CVEs(self, CVEs, expected_CVEs_path):
                     break
         assert verified_CVEs == len(expected_CVEs_path), "not all CVEs were verified"
 
+    def validate_expected_CVEs_summaries(self, CVEs_summaries, expected_CVEs_summaries_path):
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/helm/base_vulnerability_scanning.py:1068]: let's not receive a file path on the function. It should receive summaries and expected summaries and compare them. Reading from files it's not the responsibility of this function

File: tests_scripts/helm/base_vulnerability_scanning.py [hunk 8]
```diff
@@ -1035,6 +1065,49 @@ def validate_expected_CVEs(self, CVEs, expected_CVEs_path):
...
+        verified_CVEs = 0
+        if TestUtil.get_arg_from_dict(self.test_driver.kwargs, statics.CREATE_TEST_FIRST_TIME_RESULTS, False):
+            for expected_CVE_summary in expected_CVEs_summaries_path:
+                for CVE_summary in CVEs_summaries:
+                    self.store_summary_data_for_first_time_results(result=CVE_summary, store_path=expected_CVE_summary)
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/helm/base_vulnerability_scanning.py:1073]: what is this? why is it necessary? please add comments

File: tests_scripts/helm/base_vulnerability_scanning.py [hunk 9]
```diff
@@ -1035,6 +1065,49 @@ def validate_expected_CVEs(self, CVEs, expected_CVEs_path):
...
+                if CVE_summary[1]['metadata']['name'] == expected_CVE_data['metadata']['name']:
+                    for key in expected_CVE_data['metadata']['labels']:
+                        assert expected_CVE_data['metadata']['labels'][key] == CVE_summary[1]['metadata']['labels'][key], "validate_expected_CVEs_summaries: labels should be equals for key {key}. {expected} {real}".format(key=key, expected=expected_CVE_data['metadata']['labels'][key], real=CVE_summary[1]['metadata']['labels'][key])
+                    assert expected_CVE_data['spec']['severities'] == CVE_summary[1]['spec']['severities'], "the number of severities sould be equal {expected} {real}".format(excepted=expected_CVE_data['spec']['severities'], real=CVE_summary[1]['spec']['severities'])
+                    verified_CVEs += 1
+                    break
+        assert verified_CVEs == len(expected_CVEs_summaries_path), "not all CVEs summaries were verified"
+
+
+    def validate_expected_namespace_CVEs_summaries(self, CVEs_summaries, expected_CVEs_summaries_path, namespace):
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/helm/base_vulnerability_scanning.py:1091]: same as above

File: tests_scripts/helm/base_vulnerability_scanning.py [hunk 10]
```diff
@@ -1035,6 +1065,49 @@ def validate_expected_CVEs(self, CVEs, expected_CVEs_path):
...
+        verified_CVEs = 0
+        if TestUtil.get_arg_from_dict(self.test_driver.kwargs, statics.CREATE_TEST_FIRST_TIME_RESULTS, False):
+            for expected_CVE_summary in expected_CVEs_summaries_path:
+                for CVE_summary in CVEs_summaries:
+                    self.store_namespace_summary_data_for_first_time_results(result=CVE_summary, store_path=expected_CVE_summary, namespace=namespace)
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/helm/base_vulnerability_scanning.py:1096]: same as above

File: tests_scripts/helm/base_vulnerability_scanning.py [hunk 11]
```diff
@@ -1035,6 +1065,49 @@ def validate_expected_CVEs(self, CVEs, expected_CVEs_path):
...
+            for CVE_summary in CVEs_summaries:
+                with open(expected_CVE_summary[1], 'r') as content_file:
+                    content = content_file.read()
+                expected_CVE_data = json.loads(content)
+                    
+                Logger.logger.info("CVE_summary[1]['metadata']['name']: {} =?= expected_CVE_data['metadata']['name']: {}".format(CVE_summary[1]['metadata']['name'], expected_CVE_data['metadata']['name']))
+                if CVE_summary[1]['metadata']['name'] == namespace:
+                    assert expected_CVE_data['spec']['severities'] == CVE_summary[1]['spec']['severities'], "the number of severities sould be equal {expected} {real}".format(excepted=expected_CVE_data['spec']['severities'], real=CVE_summary[1]['spec']['severities'])
+                    verified_CVEs += 1
+                    break
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/helm/base_vulnerability_scanning.py:1108]: aren't we supposed to compare `vulnerabilitiesRef` as well?

File: tests_scripts/helm/base_vulnerability_scanning.py [hunk 12]
```diff
@@ -1035,6 +1065,49 @@ def validate_expected_CVEs(self, CVEs, expected_CVEs_path):
...
+                for CVE_summary in CVEs_summaries:
+                    self.store_namespace_summary_data_for_first_time_results(result=CVE_summary, store_path=expected_CVE_summary, namespace=namespace)
+                    
+        for expected_CVE_summary in expected_CVEs_summaries_path:
+            for CVE_summary in CVEs_summaries:
+                with open(expected_CVE_summary[1], 'r') as content_file:
+                    content = content_file.read()
+                expected_CVE_data = json.loads(content)
+                    
+                Logger.logger.info("CVE_summary[1]['metadata']['name']: {} =?= expected_CVE_data['metadata']['name']: {}".format(CVE_summary[1]['metadata']['name'], expected_CVE_data['metadata']['name']))
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/helm/base_vulnerability_scanning.py:1104]: please remove this / write a more readable log

File: tests_scripts/helm/relevant_cve.py [hunk 13]
```diff
@@ -68,6 +68,18 @@ def start(self):
         # 3.8 test filtered CVEs created as expected result in the storage
         self.validate_expected_filtered_CVEs(filteredCVEs, self.test_obj["expected_filtered_CVEs"], namespace=namespace)
 
+        # 3.9 test CVES summaries created as expected result in the storage
```
Comment (Daniel-GrunbergerCA) [inline tests_scripts/helm/relevant_cve.py:71]: please add info logs as we have in other tests

**Review by Daniel-GrunbergerCA — CHANGES_REQUESTED** [review]

File: tests_scripts/kubernetes/base_k8s.py (hunk 6 above)
Comment (rcohencyberarmor) [reply tests_scripts/kubernetes/base_k8s.py:1100]: same as above

File: tests_scripts/helm/base_vulnerability_scanning.py (hunk 7 above)
Comment (rcohencyberarmor) [reply tests_scripts/helm/base_vulnerability_scanning.py:1068]: we already read from file in all of our validation against expected result, so I want to keep up the same behavior as our other validations function

File: tests_scripts/helm/base_vulnerability_scanning.py (hunk 9 above)
Comment (rcohencyberarmor) [reply tests_scripts/helm/base_vulnerability_scanning.py:1091]: look the comment above

File: tests_scripts/kubernetes/base_k8s.py (hunk 3 above)
Comment (Daniel-GrunbergerCA) [reply tests_scripts/kubernetes/base_k8s.py]: You still can make it a single list right? The fact that we need to check whether it's a list or a list of lists makes the function hard to reason about

**Pull request closed without merging**